### PR TITLE
[PWA] rename components and move app specific code to /src

### PR DESCRIPTION
**Pull request by pierreneter**
Related to #1400 
Finished the first two lines in this checklist.

### CODE REVIEW

**Comment by skipjack** [comment]
@pierreneter thanks so much for tackling this! This will take a significant amount of testing/review to be sure everything is working properly. I will start testing this out as soon as I have time.

Once we've approved the changes, we can resolve the conflicts and get this merged.

cc @bebraw @hulkish

**Comment by skipjack** [comment]
Ok started testing this out locally...

The `npm run build` script runs fine, however when running `npm run lint` I get a bunch of these errors from the `lint:links` method:

``` bash
should respond with HTTP status 200
404 https://github.com/webpack/webpack.js.org/edit/master/src/content/support/index.md at build/support/index.html:19:31179 <a class="page-links__link" href="https://github.com/webpack/webpack.js.org/edit/master/src/content/support/index.md">...</a>
```

which seems to be caused by links picking up the `.md` extension when they shouldn't. I'll keep testing more and see if I can dig anything else up.

**Comment by pierreneter** [comment]
@skipjack This error occurred because I have moved the `content` and in `src` folder.
```
https://github.com/webpack/webpack.js.org/edit/master/src/content/support/index.md //after merged
https://github.com/webpack/webpack.js.org/edit/master/content/support/index.md //now
```
We should move it inwards or not?
I want to take the next steps but still waiting for this PR to be merged.
If these two steps are not necessary then we should discuss whether to merge it or not to avoid large file conflicts.

**Comment by skipjack** [comment]
@pierreneter ah you're totally right, the edit link failures are to be expected.

> We should move it inwards or not?

No it's fine as is, this is to be expected in scenarios like this.

> I want to take the next steps but still waiting for this PR to be merged.
> If these two steps are not necessary then we should discuss whether to merge it or not to avoid large file conflicts.

Can carefully resolve all the conflicts? From there I'll do one more round of testing and then make it a priority to get this merged.

**Comment by pierreneter** [comment]
All done @skipjack. Please check it soon, thank you.

**Comment by pierreneter** [comment]
all done @skipjack

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ language: node_js
 node_js:
   - "6"
 script:
-  - bash ./scripts/deploy.sh
+  - bash ./src/scripts/deploy.sh
```
Comment (skipjack) [inline .travis.yml]: We may want to keep the `/scripts` directory as is seeing as it's not part of the actual site code. @bebraw what do you think?

File: src/components/Footer/Footer.jsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import Link from '../link/link';
-import Container from '../container/container';
+import Link from '../Link/Link';
+import Container from '../Container/Container';
```
Comment (skipjack) [inline src/components/Footer/Footer.jsx]: Now that each file is named similarly just with a different extension, is there a potential for conflict here? Seeing as the site works, I'm guessing webpack tries `.jsx` files before `.scss` but isn't it somewhat ambiguous to which module this `import` goes? I do see that you did update all the styling `import`s to hit the `.scss` file directly. If this is common practice then I'm fine with it, just curious.

**Comment by skipjack** [comment]
Aside from the `lint:social` errors and edit link errors everything seems to run perfectly. I'd like to give @bebraw another day or two to review before merging but I'll hold off on merging other PRs until we hear from him so you don't need to keep rebasing.

File: .travis.yml (hunk 1 above)
Comment (skipjack) [inline .travis.yml]: Ok I'm going to leave this for now but may revert this change in a follow up PR. It's nothing major but I think it's good to separate build process related scripts from actual the actual javascript that runs the site (though the line is somewhat blurry for static site generators).

File: src/components/Footer/Footer.jsx (hunk 2 above)
Comment (skipjack) [inline src/components/Footer/Footer.jsx]: Ok still interested in thoughts on this ambiguity but as far as I can tell this isn't causing any issues.

**Comment by skipjack** [comment]
@pierreneter did a final review, pushed some fixes, and will merge shortly.

**Comment by pierreneter** [comment]
@skipjack thanks!

**Comment by skipjack** [comment]
@pierreneter had to fix one thing in #1538 but aside from that everything seems 👍  (at least no "the site blew up" issues popped up so far 😆 ).

Thanks again for your work on this, you may want to sync up with @hulkish in #1400 to discuss next steps. Note that I did check the first two items there off and added references to these PRs. It seems the other items there won't be as big overhauls but please feel free to ping me if I can help in anyway. I'm finding the PWA items, performance improvements, and lighthouse testing all very interesting.

**Pull request merged**